src/aggregator/aggregator/aggregator.go

```diff
@@ -275,8 +276,15 @@ func (agg *aggregator) AddTimedWithStagedMetadatas(
 	sw := agg.metrics.addTimed.SuccessLatencyStopwatch()
 	agg.updateStagedMetadatas(metas)
 	agg.metrics.timed.Inc(1)
-	shardingId, _ := parsers.GetMetricIDWithoutLe(metric.ID)
+	shardingId, isHistogram := parsers.GetMetricIDWithoutLe(metric.ID)
 	shard, err := agg.shardFor(shardingId)
+
+	if isHistogram {
+		agg.metrics.histogramSupportScope.Tagged(map[string]string{
+			"shard": fmt.Sprintf("%d", shard.shard),
+		}).Counter("shardCounter").Inc(1)
+		agg.logger.Debug("forwarding histogram metric from aggregator to aggregator shard", zap.Uint32("shardId", shard.shard), zap.ByteString("metricId", metric.ID))
+	}
```

Comment:
Can we add this metric to `aggregatorMetrics`? It may be nontrivial to create an object on the fly.

Reply:
I think we can't add it to `aggregatorMetrics` struct at init time since we don't know all the possible values of `shard`.  We would need to create a new counter for every possible value of `shard`, which i dont think is possible at the init time, we'd need to do it on the fly.

Reply:
You are right. I didn't notice the shard is not static here.